client/signup/config/flows.js

```diff
@@ -238,11 +239,72 @@ const Flows = {
 			flow = filterDesignTypeInFlow( flow );
 		}
 
-		return flow;
+		return Flows.getABTestFilteredFlow( flowName, flow );
 	},
 
 	getFlows() {
 		return flows;
+	},
+
+	/**
+	 * Preload AB Test variations after a certain step has been completed.
+	 *
+	 * This gives the option to set the AB variation as late as possible in the
+	 * signup flow.
+	 *
+	 * Currently only the `main` flow is whitelisted.
+	 *
+	 * @param {String} flowName The current flow
+	 * @param {String} stepName The step that is being completed right now
+	 */
+	preloadABTestVariationsForStep( flowName, stepName ) {
```

Comment:
I know this is still in-progress and I'm looking early. But it seems like we can simplify this a bit. Rather than injecting the new step in this `preloadABTestVariationsForStep()` method, we could just use `abtest()` to set and return the necessary variation in `getABTestFilteredFlow()`. That way you don't have to mirror the logic in two places. You just need to add a parameter to `getFlow()` to pass in the current stepname so it's available to pass into `Flows.getABTestFilteredFlow()`.